srp_daemon/srp_daemon.c

```diff
@@ -2052,12 +2053,16 @@ int main(int argc, char *argv[])
 			       data_details.ntc_64_67.gid) == 16);
 
 	STATIC_ASSERT(__alignof__(union umad_gid) == 4);
+	STATIC_ASSERT(offsetof(struct umad_class_port_info,
+			       redirgid) == 8);
+	STATIC_ASSERT(offsetof(struct umad_class_port_info,
+			       trapgid) == 40);
 #endif
 	STATIC_ASSERT(sizeof(struct srp_dm_mad) == 256);
 	STATIC_ASSERT(sizeof(struct srp_dm_rmpp_sa_mad) == 256);
 	STATIC_ASSERT(sizeof(struct srp_sa_node_rec) == 108);
 	STATIC_ASSERT(sizeof(struct srp_sa_port_info_rec) == 58);
-	STATIC_ASSERT(sizeof(struct srp_class_port_info) == 72);
+	STATIC_ASSERT(sizeof(struct umad_class_port_info) == 72);
 	STATIC_ASSERT(sizeof(struct srp_dm_iou_info) == 132);
 	STATIC_ASSERT(sizeof(struct srp_dm_ioc_prof) == 128);
```

Comment:
I would suggest putting these things that apply to umad structs somewhere in umad, eg a compile_test.c file that is always built.